shared/teams/confirm-modals/confirm-kick-out.tsx

```diff
@@ -4,71 +4,63 @@ import * as Container from '../../util/container'
   const members = Container.getRouteProps(props, 'members', [])
   const teamID = Container.getRouteProps(props, 'teamID', Types.noTeamID)
   const [subteamsToo, setSubteamsToo] = React.useState(false)
 
-  const [subteams, subteamIDs] = Container.useSelector(state => getSubteamNames(state, teamID))
   const teamname = Container.useSelector(state => Constants.getTeamMeta(state, teamID).teamname)
-  const waitingKeys = ([] as string[]).concat.apply(
-    members.map(member => Constants.removeMemberWaitingKey(teamID, member)),
-    members.map(member => subteamIDs.map(subteamID => Constants.removeMemberWaitingKey(subteamID, member)))
-  )
-  const waiting = Container.useAnyWaiting(...waitingKeys)
 
   const dispatch = Container.useDispatch()
   const nav = Container.useSafeNavigation()
   const onCancel = React.useCallback(() => dispatch(nav.safeNavigateUpPayload()), [dispatch, nav])
 
-  // TODO(Y2K-1592): do this in one RPC
-  const onRemove = () => {
-    dispatch(
-      TeamsGen.createTeamSetMemberSelected({
-        clearAll: true,
-        selected: false,
-        teamID: teamID,
-        username: '',
-      })
+  const removeMembersRPC = Container.useRPC(RPCTypes.teamsTeamRemoveMembersRpcPromise)
+  const [result, setResult] = React.useState<RPCTypes.TeamRemoveMembersResult>({failures: []})
+  const [waiting, setWaiting] = React.useState(false)
+  const [error, setError] = React.useState<RPCError | null>(null)
+  const removeMembers = React.useCallback(() => {
+    setWaiting(true)
+    const teamMemberToRemoves = members.map(member => {
+      return {
+        type: 0, // TODO how do I get RPCTypes.TeamMemberToRemoveType.assertion to typecheck?
+        assertion: {
+          assertion: member,
+          removeFromSubtree: subteamsToo,
+        },
+      }
+    })
+    removeMembersRPC(
+      [{teamID, members: teamMemberToRemoves, noErrorOnPartialFailure: true}],
+      result => {
+        setResult(result)
+        setWaiting(false)
+      },
+      err => {
+        setError(err)
+        setWaiting(false)
+      }
     )
+  }, [teamID, members, removeMembersRPC, subteamsToo, waiting, setWaiting])
 
-    members.forEach(member =>
-      dispatch(
-        TeamsGen.createRemoveMember({
-          teamID,
-          username: member,
-        })
-      )
-    )
-    if (subteamsToo) {
-      subteamIDs.forEach(subteamID =>
-        members.forEach(member =>
-          dispatch(TeamsGen.createRemoveMember({teamID: subteamID, username: member}))
-        )
-      )
-    }
+  const onRemove = () => {
+    removeMembers()
   }
 
+  const anyError = error || result?.failures?.length
+
   const wasWaiting = Container.usePrevious(waiting)
   React.useEffect(() => {
-    if (wasWaiting && !waiting) {
-      dispatch(RouteTreeGen.createNavUpToScreen({routeName: 'team'}))
+    if (wasWaiting && !waiting && !anyError) {
+      // TODO this breaks the app, [team member view] no data! this should never happen
```

Comment:
question, this is completely wrong in the success case but i couldn't figure out how to fix it (making sure it works for all the places where the modal can be called, chat, team page, member page)